smart-contract-verifier/smart-contract-verifier/src/sourcify/types.rs

```diff
@@ -35,10 +35,81 @@ pub struct Success {
     pub match_type: MatchType,
 }
 
+impl TryFrom<sourcify::GetSourceFilesResponse> for Success {
+    type Error = Error;
+
+    fn try_from(value: sourcify::GetSourceFilesResponse) -> Result<Self, Self::Error> {
+        let metadata: ethers_solc::artifacts::Metadata =
+            serde_json::from_value(value.metadata.clone()).map_err(|err| {
+                tracing::error!(target: "sourcify", "returned metadata cannot be parsed: {err}");
+                Error::Internal(anyhow::anyhow!(
+                    "error occurred when parsing sourcify response"
+                ))
+            })?;
+
+        // Compiler settings inside metadata contains a "compilationTarget"
+        // which does not exist in compiler input. We should remove the key
+        // to make the settings which could be used for the compiler input.
+        let compiler_settings = {
+            let mut compiler_settings = value
+                .metadata
+                .as_object()
+                .expect("metadata has been parsed successfully and must be an object")
+                .get("settings")
+                .expect("metadata has been parsed successfully and must contain 'settings' key")
+                .as_object()
+                .expect("metadata has been parsed successfully and 'settings' must be an object")
+                .clone();
+
+            compiler_settings.remove("compilationTarget");
+            compiler_settings
+        };
+
+        let evm_version = compiler_settings
+            .get("evmVersion")
+            .and_then(|value| value.as_str().map(|value| value.to_string()));
+
+        let abi = value
+            .metadata
+            .as_object()
+            .expect("metadata has been parsed successfully and must be an object")
+            .get("output")
+            .expect("metadata has been parsed successfully and must contain 'output' key")
+            .as_object()
+            .expect("metadata has been parsed successfully and 'output' must be an object")
+            .get("abi")
+            .expect("metadata has been parsed successfully and must contain 'output.abi' key")
```

Comment:
I suggest to parse metadata to stucture here. or maybe even not here but in sourcify lib